### ISSUE TITLE
Support show on winbar

### ISSUE THREAD

**Issue by leaxoy** [SmiteshP/nvim-gps#99]
https://github.com/neovim/neovim/pull/17336 winbar ui has been merged into nightly, it's general to show the gps component, there is vscode example:
<img width="815" alt="image" src="https://user-images.githubusercontent.com/8556349/169230951-bcf60900-874f-4c5b-8ce1-6c7944a551e8.png">
and neovim example: 
![image](https://user-images.githubusercontent.com/8556349/169231034-178da62a-514c-4856-a904-9d41b0663e91.png)

**Comment by rockyzhang24** [comment]
@wookayin Thanks for the code. However, `winbar` won't be updated automatically as `statusline` when we move cursor, so it seems we need to use `autocmd CursorMoved * ` to update it.

**Comment by wookayin** [comment]
> However, `winbar` won't be updated automatically as `statusline` when we move cursor, so it seems we need to use `autocmd CursorMoved * ` to update it.

@rockyzhang24 No, that's not correct and the above code **works** as you'd expect. Did you try them out?`%v` expression means that the function will be called whenever needed (including after cursor update). Read the documentation:`:help statusline`

**Comment by rockyzhang24** [comment]
@wookayin confirmed that this is a bug of the newly added `winbar`(see this https://github.com/neovim/neovim/pull/17336#issuecomment-1132085879). I know it works as expected for statusline.

**Comment by leaxoy** [comment]
Corner case, the winbar always shows even gps is not available.

Correct case
<img width="590" alt="image" src="https://user-images.githubusercontent.com/8556349/169448654-a5c513d1-5a9e-41d2-aebe-19c19af24b06.png">

Wrong case
<img width="639" alt="image" src="https://user-images.githubusercontent.com/8556349/169448699-a46ae789-851c-4ef3-a5d1-e4639cddea5e.png">

**Comment by rockyzhang24** [comment]
@leaxoy Use my code above and it works well. Because winbar won't be updated when you move your cursor, thus you need the autocmd to refresh it. When that mentioned issue gets fixed, you can remove the autocmd part.

**Comment by rockyzhang24** [comment]
@emuel-vassallo Using the autocmd is just a temporary workaround to solve the winbar update issue I mentioned above and it causes this telescope side effect. Waiting for the fix (https://github.com/neovim/neovim/pull/18646) to be merged, and then the autocmd will not be needed.

**Comment by rockyzhang24** [comment]
@emuel-vassallo Try to change the autocmd like this to avoid it disrupting telescope. 
```lua
vim.api.nvim_create_autocmd ("CursorMoved", {
  pattern = '*',
  command = "redrawstatus",
})
```

**Comment by emuel-vassallo** [comment]
> @emuel-vassallo Try to change the autocmd event to `CursorHold` to avoid it disrupting telescope windows.

@rockyzhang24 That works, thank you!

**Comment by SmiteshP** [comment]
Will add instructions in README on how to setup in winbar 👍🏽

**Comment by rockyzhang24** [comment]
@SmiteshP Great! You can wait until that PR fixing the update bug gets merged.

**Comment by max397574** [comment]
it's merged

**Comment by rockyzhang24** [comment]
Great. Then adding gps to winbar is the same with adding it to statusline. It's very handy.

**Comment by nghialm269** [comment]
Now that https://github.com/neovim/neovim/pull/18650 is merged, is it possible to add click support to each part of gps to jump to it?

**Comment by siduck** [comment]
https://github.com/fgheng/winbar.nvim
